locksmith/src/controllers/lockController.js

```diff
@@ -36,6 +42,57 @@ const lockGet = async (req, res) => {
   res.json(baseTokenData)
 }
 
+migrateLogEvent.on('migrateLock', async ({ recordId, msg }) => {
+  console.log('migrateLogEvent', recordId, msg)
+  await updateLockMigrationsLog(recordId, msg)
+})
+
+const lockMigrate = async (req, res) => {
+  const { lockAddress } = req.params
+  const unlockVersion = req.query.unlockVersion || 9
+  const chainId = req.chain
+
+  // TODO: make sure lock exists
+  // const databaseLock = await getLockByAddress(lockAddress)
+  // if (!databaseLock) res.send(404)
+
+  // record the migration in db
+  const dbRecord = await LockMigrations.create({
+    lockAddress,
+    // initiatedBy: msg.sender // TODO: get that info somewhere?
```

Comment:
Initially I was thinking we should ask the user to sign a message  but I don't think this is required and I worry this adds complexity?

Reply:
As you noted I think good enough to use a query string populated by front-end?

Reply:
yep ok for using URL param